keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -483,25 +489,48 @@ class ClusterHandler(
     }
 
     // TODO: Frigga and Rocket version parsing are not aligned. We should consolidate.
-    val artifactName = checkNotNull(base.launchConfiguration.appVersion).parseAppVersion().packageName
-
-    val status = debianArtifactParser.parseStatus(base.launchConfiguration.appVersion?.substringAfter("$artifactName-"))
-    if (status == UNKNOWN) {
-      throw ExportError("Unable to determine release status from appVersion ${base.launchConfiguration.appVersion}, you'll have to configure this artifact manually.")
+    val appVersion = checkNotNull(base.launchConfiguration.appVersion).parseAppVersion()
+    val artifactName = appVersion.packageName
+    val artifactVersion = base.launchConfiguration.appVersion!!.removePrefix("${artifactName}-")
+    val artifact = try {
+      artifactService.getArtifact(artifactName, artifactVersion, DEBIAN)
```

Comment:
Here's where we try to retrieve basic artifact information from Igor for the artifact, which we use to fill in the branch in the artifact constructor below.